lib/Mail/Pyzor/Client.pm

```diff
@@ -348,16 +348,16 @@ sub _send_packet {
 
 sub _get_connection_or_die {
     my ($self) = @_;
-    $self->{'_sock'} ||= IO::Socket::INET->new(
+    my $sock = IO::Socket::INET->new(
```

Comment:
Using the PID to invalidate the cache would seem a more performant way of fixing this:
```
if ($self->{'_sockpid'} && $$ != $self->{'_sockpid'}) {
    undef $self->{'_sock'};
}
$self->{'_sock'} ||= IO::Socket::INET->new(
...
```

Reply:
I agree